### ISSUE TITLE
[melonDS] Microphone support

### ISSUE THREAD

**Issue by medigi** [TASVideos/BizHawk#1914]
There is some games, where is microphone can be used or need to be used:

Rayman - Raving Rabbids
Garfield - A Tail of Two Kitties
WarioWare Gold

And so on

**Comment by Spikestuff** [comment]
You're correct that it doesn't have support for microphones... for the time being.
Just to add in a recent IRC convo about this topic:

<adelikat> so what does the melonDS core need to support, if we go with this playlist thing?
<zeromus> the frontend feature, to be clear, would work this way. "mic-rec: hold this key to record a new mic sample to replace your prior one. this is unavailable if a mic-list is loaded."; "mic-play: hold this key to play the current mic sample, either just recorded or loaded from the mic-list." "mic-next/prev: select next/prev sample from mic-list". "mic-list-load: select a mic list file"; 
<zeromus> "mic-list-clear: deselect the mic list file"
<zeromus> melonds needs to support a "set mic sample" command and a "mic-play hold" virtual button.

**Comment by SuuperW** [comment]
For anybody who wants to implement this, libmelonDS.dll exposes a method `void SetMicSamples(short* samples, int count)` which sets the mic samples to be used on the next frame.

For the record, I disagree with the playlist idea mentioned in the above post. It's a hackish way of avoiding the work of including a new input type which I think will cause more issues and confusion than implementing a byte stream input type.

**Comment by getCursorsExe** [comment]
Well, "Mario and Luigi: Bowser's Inside Story" can be played, but the game will softlock during the first Giant Bowser battle.
The softlock occurs when Spinies will appear next to the castle.
